### PR TITLE
Update kaldi version

### PR DESCRIPTION
**Pull request by jcftang**
The kaldi build system is a bit messy as well as the fact that they don't do tags in their repo, this was the best I could come up with for fixing the package. 

I think this package probably needs a few more iterations of adding versions before a clean up can be done.

### CODE REVIEW

File: var/spack/repos/builtin/packages/kaldi/package.py [hunk 1]
```diff
@@ -85,7 +91,11 @@ def install(self, spec, prefix):
                         install(join(root, name), prefix.bin)
 
             mkdir(prefix.lib)
-            install_tree('lib', prefix.lib)
+            for root, dirs, files in os.walk('lib'):
+                for name in files:
+                    if name.endswith(".so"):
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/kaldi/package.py]: What about on macOS? Might want to use `dso_suffix` instead.

**Comment by weijianwen** [comment]
This PR is duplicate with #12177 .

**Comment by jcftang** [comment]
It's not quite a duplicate but it builds on that commit, i wanted to keep fixing the package and updating it with a newer version separate.

**Comment by scheibelp** [comment]
Hi @jcftang https://github.com/spack/spack/pull/12177 was merged and has created a conflict here. When you get a chance can you resolve that?

**Comment by adamjstewart** [comment]
Closing since we got no response, and this PR seems like it is superseded by #12177 and #14398

**Pull request closed without merging**